### DISCUSSION TITLE
Trouble verifying if graphical issues are emulation related

### DISCUSSION THREAD

**Discussion by chroniceel93** [86Box/86Box#4469] [Q&A] [answered]
I've been on a bit of a tear messing around with a Voodoo 3, Pentium MMX system running Windows 98, installing various games I've both got nostalgia for, and games I've always wanted to. Of course, most of these games are very late 90's, early 00's games, right on the edge of what this sort of system is meant to run. In my time doing this, I've come across several different issues that appear to be present on actual hardware, and some that I suspect might be emulation related.
In particular, I've got two interesting cases I can't really verify. First off is American McGee's Alice.
![Monitor_1_20240520-183100-832](https://github.com/86Box/86Box/assets/37881376/de5fdd14-445e-4215-912c-ffa70a5d2448)
Now I wouldn't say this game runs terrifically by any measure, but it's certainly playable. The only, and most glaring issue, seems to be the UI.
The second problem program is Airfix Dogfighter, a game I played _incessantly_ when I was a kid. It seems to vary between hanging Windows, and hanging 86Box itself. I'm mostly sure this one's an emulation issue.
That said, for both of these games, I don't have original hardware to test this on, and I'm not sure how to verify whether either of these are emulation issues.
I've got at least one bug that's emulated _correctly_ to my great amusement. Harry Potter and the Sorcerer's stone is a game that only barely runs on this hardware, and you have to trick it a little to make it use the voodoo card as a 3d accelerator, but when you do, you'll find that the textures are all blurred. I'm pretty sure this is the mipmapping breaking, and I've verified elsewhere that this really was a problem with Voodoo3 cards.
![image](https://github.com/86Box/86Box/assets/37881376/c965facc-1bf2-408a-9de9-960a301f657c)

**Comment by hkevinli** [comment]
For Alice, your best bet is to use MesaFX. 3Dfx's opengl ICD has problems with UI elements.

For the blurry textures in Harry Potter, try setting LOD bias to -2.

**Reply by chroniceel93** [reply, marked as answer]
MesaFX fixed Alice right up. I've tried setting the LOD bias for Harry Potter before, and it didn't work. Did it again just now to verify it still doesn't work. I found [_one_](https://falconfly.vogonswiki.com/cgi-bin/yabb2/YaBB93b0.html?num=1031351068/0) forum post displaying that behavior on real hardware, with a similar system from decades ago that never had a solution.
